packages/chat/src/components/VoiceWidget/WaveformVisualizer.component.tsx

```diff
@@ -0,0 +1,92 @@
+import { useEffect, useRef } from 'react';
+
+export const WaveFormVisualizer = () => {
+  const canvasRef = useRef<HTMLCanvasElement>(null);
+  const audioContextRef = useRef<AudioContext | null>(null);
+  const analyserRef = useRef<AnalyserNode | null>(null);
+  const dataArrayRef = useRef<Uint8Array | null>(null);
```

Comment:
same here, we need only canvasRef, everything else can be stored as local variables in the effect...